### ISSUE TITLE
When opening the app, if multiple SSH tabs were open before, it asks to unlock the key for each tab.

### ISSUE THREAD

**Issue by joshuamars** [elementary/files#2052]
### What Happened?

I have a server that I was connected to with SSH. I had three tabs opened for this connection. After closing the app and restarting my computer, I opened the app and was asked several times to unlock the same key for each tab.

Using version 6.1.0

### Steps to Reproduce

1. Open a new SSH connection by clicking Connect Server
2. Enter details and connect (I use a password protected key for this connection, so I leave the password blank)
3. Open multiple tabs to locations on that server
4. Close the Files app and log out (to lock the key)
5. Log back in and open the Files app

App now prompts you to unlock the (same) key for each of the tabs open.

### Expected Behavior

I think it only needs to ask me **once** to unlock the key right? Just closing the app or opening a new tab after the key was unlocked works fine.

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

**Comment by jeremypw** [comment]
I guess because each tab is restored asynchronously and simultaneously, so fixing this may not be straightforward but yeah ideally the key should only need unlocking once. We would have arrange for tabs pointing to the same SSH server to open sequentially I suppose.

**Comment by vjr** [comment]
What about only opening the tabs "partially" meaning only show the tab title and actually load the first visible tab such that if it's the ssh location it will prompt then switching to another tab with same ssh host will lazy load the tab so that it sees already authenticated and doesn't prompt?

I mean something like browsers do when restoring tabs? Only the first visible/active tab is loaded and the rest are lazy loaded when actually clicked to make visible.

**Comment by jeremypw** [comment]
Yes that's a possibility.

**Comment by jeremypw** [comment]
I tried to reproduce this with the latest version of Files and found that the SSH tabs were not restored. I had created bookmarks to their locations and opening one resulted in a request for a password but then no further requests were made.  However the my setup may not match that of the OP exactly.

**Comment by jeremypw** [comment]
@joshuamars  Are you able to reproduce this issue with the latest version of Files?

**Comment by joshuamars** [comment]
@jeremypw I haven't organically been in a situation where the bug arises for a while. I just tried to reproduce it, but as you experienced, SSH tabs were not automatically restored. Maybe it isn't relevant anymore. The real reason the bug bothered me is because I had to click through the prompts just to close old tabs I wasn't necessarily intending to use.

I guess, is it intended behavior for the files app not to try and reopen SSH tabs at start?

**Comment by jeremypw** [comment]
Yes, restoring remote locations is problematic when the connection or server may be down so it is easier not to restore them unless we can handle those situations gracefully.
